scripts/build/builders/tizen.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+
+from enum import Enum, auto
+
+from .builder import Builder
+
+
+class TizenApp(Enum):
+    LIGHT = auto()
+
+    def ExampleName(self):
+        if self == TizenApp.LIGHT:
+            return 'lighting-app'
+        else:
+            raise Exception('Unknown app type: %r' % self)
+
+    def AppNamePrefix(self):
+        if self == TizenApp.LIGHT:
+            return 'chip-tizen-lighting-example'
+        else:
+            raise Exception('Unknown app type: %r' % self)
+
+
+class TizenBoard(Enum):
+    ARM = auto()
+
+    def TargetCpuName(self):
+        if self == TizenBoard.ARM:
+            return 'arm'
+        else:
+            raise Exception('Unknown board type: %r' % self)
+
+
+class TizenBuilder(Builder):
+
+    def __init__(self,
+                 root,
+                 runner,
+                 output_prefix: str,
+                 app: TizenApp = TizenApp.LIGHT,
+                 board: TizenBoard = TizenBoard.ARM):
+        super(TizenBuilder, self).__init__(
+            root=os.path.join(root, 'examples', app.ExampleName(), 'linux'),
+            runner=runner,
+            output_prefix=output_prefix)
+        self.app = app
+        self.board = board
+
+    def generate(self):
+        if not os.path.exists(self.output_dir):
+            if not self._runner.dry_run:
+                if 'TIZEN_HOME' not in os.environ:
+                    raise Exception(
+                        "Environment TIZEN_HOME missing, cannot build tizen libraries")
+
+            cmd = '''\
+export PKG_CONFIG_SYSROOT_DIR=$TIZEN_HOME;
+export PKG_CONFIG_LIBDIR=$TIZEN_HOME/usr/lib/pkgconfig;
```

Comment:
The build should not depend on these environment variables being set externally via python.

@erjiaqing Is adding the sysroot configs. This should be passed to the pkg-config wrapper's sysroot option via the pkg_config() template not environment variables.

@erjiaqing - ETA for the PR for this?

Reply:
The pkg_config.py already supports sysroot.

Reply:
pkg_config.py only set 'PKG_CONFIG_LIBDIR' in now. Did not set 'PKG_CONFIG_SYSROOT_DIR' environment variable.
And, basically we refer on next custom build PR for Tizen builder. In there, mention to set it before gn build.
https://github.com/project-chip/connectedhomeip/pull/8077

    And the pkg-config tool need below two enviroment to be well set  to
    correctly find the include path:
      * PKG_CONFIG_SYSROOT_DIR
      * PKG_CONFIG_LIBDIR

    A demo command to generate ninja files are listed below, the content in
    angle brackets need to be updated with developer's local env.

    PKG_CONFIG_SYSROOT_DIR=<embedded_linux_sdk_sysroot> \
    PKG_CONFIG_LIBDIR=${PKG_CONFIG_SYSROOT_DIR}/usr/lib/pkgconfig:${PKG_CONFIG_SYSROOT_DIR}/usr/share/pkgconfig \
    gn gen out out/aarch64 --args='target_os="linux" target_cpu="arm64" arm_arch="armv8-a"
          target_cflags=[ "--sysroot=<embedded_linux_sdk_sysroot>" ]
          target_ldflags=[ "--sysroot=<embedded_linux_sdk_sysroot>" ]
          linux_arm_cross_compile="<path_to_cross_compile_tools>"'

Reply:
I will change this option("export PKG_CONFIG_LIBDIR=$TIZEN_HOME/usr/lib/pkgconfig;") after pkg_config.gni modified.

Reply:
Regarding pkgconfig's sysroot environment, it seems to be doing the review seperately.
https://github.com/project-chip/connectedhomeip/pull/9352

We will integrate the change after applying above PR for this file.
So how about resovling this conversation?

Actually, except of this we completed to handle comments.
And we should apply remain Tizen patchsets for implementing.